src/RunspaceProviders/Kubernetes/VMware.ScriptRuntimeService.K8sRunspaceProvider/K8sRunspaceProvider.cs

```diff
@@ -497,122 +507,267 @@ public class K8sRunspaceProvider : IRunspaceProvider {
-                     Id = result.Id,
+                  return new K8sWebConsoleInfo {
+                     Id = webConsoleInfo.Id,
                      CreationState = RunspaceCreationState.Error,
                      CreationError = new RunspaceProviderException(Resources.K8sRunspaceProvider_WaitCreateComplition_TimeOut)
                   };
-               } else
-               if (HasErrrorInContainerStatus(pod.Status, out var errorMessage)) {
-                  // Container Creation Error
-                  result = new K8sRunspaceInfo {
-                     Id = result.Id,
-                     CreationState = RunspaceCreationState.Error,
-                     CreationError = new RunspaceProviderException(errorMessage)
-                  };
                } else {
                   // Success, everything should be in place
-                  result = new K8sRunspaceInfo {
-                     Id = result.Id,
-                     Endpoint =
-                        new IPEndPoint(
-                           IPAddress.Parse(pod.Status.PodIP),
-                           _runspaceApiPort),
+                  return new K8sWebConsoleInfo {
+                     Id = webConsoleInfo.Id,
                      CreationState = RunspaceCreationState.Ready
                   };
                }
-
-               if (result.CreationState == RunspaceCreationState.Ready && _verifyRunspaceApiIsAccessibleOnCreate) {
-                  try {
-                     _logger.LogDebug($"EnsureRunspaceEndpointIsAccessible: Start");
-                     // Ensure Container is accessible over the network after creation
-                     EnsureRunspaceEndpointIsAccessible(result);
-                     _logger.LogDebug($"EnsureRunspaceEndpointIsAccessible: Success");
-                  } catch (RunspaceProviderException exc) {
-                     LogException(exc);
-                     // Kill the container that is not accessible, otherwise it will leak
-                     try {
-                        Kill(result.Id);
-                     } catch (RunspaceProviderException rexc) {
-                        LogException(rexc);
-                     }
-
-                     result = new K8sRunspaceInfo {
-                        Id = result.Id,
-                        CreationState = RunspaceCreationState.Error,
-                        CreationError = exc
-                     };
-                  }
-               }
+            } else {
+               return podWaitResult;
             }
          }
-         return result;
+      }
+
+      private static bool IsNginxReloadEventAfter(Corev1Event e, DateTime since) {
+         return e.LastTimestamp?.CompareTo(since) > 0 &&
+            e.Type.Equals("Normal", StringComparison.InvariantCultureIgnoreCase) &&
+            e.Reason.Equals("RELOAD", StringComparison.InvariantCultureIgnoreCase) &&
+            e.Message.Equals("NGINX reload triggered due to a change in configuration");
```

Comment:
This `equals` seems fragile. Put in the backlog a task for creating a heartbeat API to replace this check here.

Reply:
done